controlplane/internal/controllers/rke2controlplane_controller.go

```diff
@@ -483,35 +500,17 @@ func (r *RKE2ControlPlaneReconciler) reconcileDelete(ctx context.Context, cluste
 	return ctrl.Result{RequeueAfter: deleteRequeueAfter}, nil
 }
 
-// 	// TODO: Improve this part once there are dependencies on the Control Plane Object!
-// 	var ok bool = true
-// 	for _, finalizer := range rcp.GetObjectMeta().GetFinalizers() {
-// 		if ok {
-// 			ok = controllerutil.RemoveFinalizer(rcp, finalizer)
-// 		}
-// 	}
-// 	if !ok {
-// 		logger.Info("unable to remove all finalizers")
-// 		err = fmt.Errorf("unable to remove all finalizers")
-// 		res = ctrl.Result{}
-// 		return
-// 	}
-
-// 	if err = r.Delete(ctx, rcp); err != nil {
-// 		res = ctrl.Result{RequeueAfter: 2 * time.Minute}
-// 		return
-// 	}
-
-// 	return ctrl.Result{}, nil
-// }
+// 	TODO: Improve this part once there are dependencies on the Control Plane Object!
 
 func (r *RKE2ControlPlaneReconciler) reconcileKubeconfig(
 	ctx context.Context,
 	clusterName client.ObjectKey,
 	endpoint clusterv1.APIEndpoint,
 	rcp *controlplanev1.RKE2ControlPlane) (ctrl.Result, error) {
 
+	logger := ctrl.LoggerFrom(ctx)
 	if endpoint.IsZero() {
+		logger.V(5).Info("API Endpoint not yet known")
 		return ctrl.Result{}, nil
```

Comment:
Maybe requeue after, so the API endpoint may be available before the next scheduled resync.

Reply:
Added `RequeueAfter: 30 * time.Second` , what do you think?

Reply:
Tracking a broader review of Requeue strategy on issue #82